api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -419,8 +419,8 @@ public DbWorkspace saveAndCloneCohortsConceptSetsAndDataSets(DbWorkspace from, D
   }
 
   @Override
-  public WorkspaceAccessLevel getWorkspaceAccessLevel(
-      String workspaceNamespace, String workspaceId) {
+  public WorkspaceAccessLevel getWorkspaceAccessLevel(String workspaceNamespace, String workspaceId)
+      throws ServerErrorException {
```

Comment:
My preference would be to not throw HTTP level exceptions at the service layer. We can throw more generic exceptions from here and the controller can catch that and throw the HTTP status code it believes is most relevant.

Reply:
I'd go even further than that and say we should basically never throw exceptions for predictable things like this at the service level.

However, I'm not changing any exception behavior in this PR, just denoting places where it's easy to get burned. If we do want exceptions to be a part fo the contract for service methods, then IMO those should be checked exceptions.

Putting unchecked exceptions in a throws statement like this is just a glorified comment, but I think the added clarity on what's actually happening is better than being surprised.

Now as for this function, I'm wondering if we should throw IllegalArgumentException when the access isn't recognized, or just default to `NO_ACCESS`. I think the latter is a safer thing to do, and then we can drop the throws statement, but really this is a case (programming error) where an undeclared, unchecked exception is OK. Really, the generated `WorkspaceAccessLevel.fromValue()` should do any throwing or defaulting for us, but what are you gonna do?

Reply:
Filed [this bug](https://precisionmedicineinitiative.atlassian.net/browse/RW-4105), which you could argue is a story/refactoring.